src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,498 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+            volume = storagePool.volumes.find(vol => vol.name === diskParams.existingVolumeName);
             isVolumeUsed = getStorageVolumesUsage(vms, storagePool);
         }
 
         return domainAttachDisk({
             connectionName: vm.connectionName,
-            type: this.state.mode === CUSTOM_PATH ? "file" : "volume",
-            file: this.state.file,
-            device: this.state.device,
-            poolName: this.state.storagePoolName,
-            volumeName: this.state.existingVolumeName,
-            format: this.state.format,
-            target: this.state.target,
-            permanent: this.state.permanent,
-            hotplug: this.state.hotplug,
+            type: mode === CUSTOM_PATH ? "file" : "volume",
+            file: diskParams.file,
+            device: diskParams.device,
+            poolName: storagePoolName,
+            volumeName: diskParams.existingVolumeName,
+            format: diskParams.format,
+            target: diskParams.target,
+            permanent: diskParams.permanent,
+            hotplug,
             vmName: vm.name,
             vmId: vm.id,
-            cacheMode: this.state.cacheMode,
-            shareable: volume && volume.format === "raw" && isVolumeUsed[this.state.existingVolumeName].length > 0,
-            busType: this.state.busType,
-            serial: this.state.serial
+            cacheMode: diskParams.cacheMode,
+            shareable: volume && volume.format === "raw" && isVolumeUsed[diskParams.existingVolumeName].length > 0,
+            busType: diskParams.busType,
+            serial: diskParams.serial
         })
                 .then(() => { // force reload of VM data, events are not reliable (i.e. for a down VM)
                     const promises = [];
-                    if (this.state.mode !== CUSTOM_PATH && volume.format === "raw" && isVolumeUsed[this.state.existingVolumeName]) {
-                        isVolumeUsed[this.state.existingVolumeName].forEach(vmName => {
+                    if (mode !== CUSTOM_PATH && volume.format === "raw" && isVolumeUsed[diskParams.existingVolumeName]) {
+                        isVolumeUsed[diskParams.existingVolumeName].forEach(vmName => {
                             const vm = vms.find(vm => vm.name === vmName);
-                            const diskTarget = getStorageVolumeDiskTarget(vm, storagePool, this.state.existingVolumeName);
+                            const diskTarget = getStorageVolumeDiskTarget(vm, storagePool, diskParams.existingVolumeName);
 
                             promises.push(
                                 domainUpdateDiskAttributes({ connectionName: vm.connectionName, objPath: vm.id, readonly: false, shareable: true, target: diskTarget })
-                                        .catch(exc => this.dialogErrorSet(_("Disk settings could not be saved"), exc.message))
+                                        .catch(exc => dialogErrorSet(_("Disk settings could not be saved"), exc.message))
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231120-122732-13cb53e1-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)